src/odemis/driver/test/keysight_test.py

```diff
@@ -107,15 +110,17 @@ def test_delay(self):
         """
         self.dev.period.value = 1e-4
         self.dev.delay.value = 1e-5
-        self.assertEqual(self.dev.delay.value, 1e-5)
+        self.assertAlmostEqual(self.dev.delay.value, 1e-5)
         self.dev.delay.value = 1e-6
-        self.assertEqual(self.dev.delay.value, 1e-6)
+        self.assertAlmostEqual(self.dev.delay.value, 1e-6)
+        self.dev.delay.value = -1e-6
+        self.assertAlmostEqual(self.dev.delay.value, -1e-6)
 
         with self.assertRaises(IndexError):
             self.dev.delay.value = self.dev.delay.range[1] + 1
```

Comment:
if applicable, we can have a similar test for the other value of range -> delay.range[0]  - 1